distributed/client.py

```diff
@@ -806,6 +808,150 @@ class VersionsDict(TypedDict):
     client: dict[str, dict[str, Any]]
 
 
+_T_LowLevelGraph: TypeAlias = dict[Key, tuple]
+
+
+class MapLayer(Layer):
+    # func: Callable
+    # iterable: Iterable
+    # key=None
+    # workers=None
+    # retries=None
+    # resources=None
+    # priority: int = 0
+    # allow_other_workers: bool = False
+    # fifo_timeout: str = "100 ms"
+    # actor: bool = False
+    # actors: bool = False
+    # pure: bool = True
+    # batch_size=None
+    # # **kwargs
+
+    def __init__(
+        self,
+        func: Callable,
+        iterables: Iterable,
+        key,
+        workers=None,
+        retries=None,
+        resources=None,
+        priority: int = 0,
+        allow_other_workers: bool = False,
+        fifo_timeout: str = "100 ms",
+        actor: bool = False,
+        actors: bool = False,
+        pure: bool = True,
+        batch_size=None,
+        parts_out=None,  # TODO check this
+        annotations=None,
```

Comment:
I think all of this should not be required to build the graph and should therefore not be part of the HLG object.